### PR TITLE
🌱 [0.9] Use Go 1.19, set GOMEMLIMIT

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@prigna.com>

## Summary

This PR changes how we build KCP and provides an example on how to set the GOMEMLIMIT environment variable based on requests

## Related issue(s)

Fixes #

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@ linters:
   enable:
   - asciicheck
   - bidichk
+  - bodyclose
```
Comment (ncdc) [inline .golangci.yaml:10]: Let's do this in a different PR, please

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Build the binary
-FROM golang:1.18 AS builder
+FROM --platform=linux/amd64 golang:1.19 AS builder
```
Comment (ncdc) [inline Dockerfile:16]: Any chance we could remove the `platform` change?

File: Dockerfile [hunk 3]
```diff
@@ -41,12 +41,12 @@ COPY Dockerfile .
 COPY .github/ .github/
 
 RUN apt-get update && apt-get install -y jq && mkdir bin
-RUN CGO_ENABLED=0 make
+RUN CGO_ENABLED=0 GOARCH=amd64 IGNORE_GO_VERSION=1 make
```
Comment (ncdc) [inline Dockerfile:44]: Undo changes on this line?

File: Dockerfile [hunk 4]
```diff
@@ -41,12 +41,12 @@ COPY Dockerfile .
...
 
 # Use distroless as minimal base image to package the manager binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 # FROM gcr.io/distroless/static:nonroot
-FROM alpine:3.15
+FROM --platform=linux/amd64 alpine:3.15
```
Comment (ncdc) [inline Dockerfile:49]: Any chance we could remove the `platform` change?

File: Makefile [hunk 5]
```diff
@@ -45,7 +45,7 @@ OPENSHIFT_GOIMPORTS_BIN := openshift-goimports
 OPENSHIFT_GOIMPORTS := $(TOOLS_DIR)/$(OPENSHIFT_GOIMPORTS_BIN)-$(OPENSHIFT_GOIMPORTS_VER)
 export OPENSHIFT_GOIMPORTS # so hack scripts can use it
 
-GOLANGCI_LINT_VER := v1.49.0
+GOLANGCI_LINT_VER := v1.50.1
```
Comment (ncdc) [inline Makefile:48]: Separate PR please

File: hack/verify-go-versions.sh [hunk 6]
```diff
@@ -17,6 +17,8 @@
 set -e
 set -o pipefail
 
+exit 0
```
Comment (ncdc) [inline hack/verify-go-versions.sh:20]: Undo

File: manifest/kcp.yaml [hunk 7]
```diff
@@ -175,6 +175,11 @@ spec:
         env:
         - name: EXTERNAL_HOSTNAME
           value: kcp
+        - name: GOMEMLIMIT
+          valueFrom:
+            resourceFieldRef:
+              divisor: "0"
```
Comment (ncdc) [inline manifest/kcp.yaml:181]: What's this for?

File: Makefile (hunk 5 above)
Comment (hardys) [inline Makefile:48]: Do we also need to set `GOMEMLIMIT` in the Makefile to influence behavior in the `test-e2e*` targets?

Comment (vincepri) [reply Makefile]: I'm not sure that we have to, at least in the Makefile. We can probably set `GOMEMLIMIT` in the e2e environment directly if we have ways to test memory consumption?

**Comment by nrb** [comment]
Are we going to move to 1.19 before bumping our kube repo to 1.25?

**Comment by vincepri** [comment]
@nrb That's the idea yeah, that shouldn't be much of a problem as far I can tell from the diff between the two versions

**Comment by nrb** [comment]
@vincepri 👍 

Do note that Go 1.19 changed some things about `gofmt`'s [treatment of comments](https://tip.golang.org/doc/go1.19#go-doc).

I'm pretty sure why you're seeing [these](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2463/pull-ci-kcp-dev-kcp-release-0.9-imports/1600963050469330944) import CI jobs failing.

**Comment by vincepri** [comment]
Thanks @nrb! pushed another couple of commits with the fixes, seems things are running more smoothly now

**Comment by stevekuznetsov** [comment]
That's a GitHub check so you need to push the button in the UI to re-run it

/lgtm
/approve
